lib/rules/no-unused-vars.js

```diff
@@ -711,8 +789,25 @@ module.exports = {
                         });
                     }
                 }
+
+                if (config.reportUsedIgnorePattern) {
+                    for (const usedIgnoredVar of usedIgnoredVars) {
+                        if (usedIgnoredVar.defs.length > 0) {
+                            const references = usedIgnoredVar.references.filter(
+                                ref => ref.isRead() && ref.from.variableScope === usedIgnoredVar.scope.variableScope
```

Comment:
Per this condition, `_a` wouldn't be reported at all in the following example, which seems wrong as it's used?

```js
/* eslint no-unused-vars: ["error", {
    varsIgnorePattern: "^_",
    reportUsedIgnorePattern: true
}]*/

var _a = 42;

foo(() => _a);

```

Reply:
Do you think I should revert to just `ref => ref.from.variableScope === usedIgnoredVar.scope.variableScope`? Is there a better condition to use?